### PR TITLE
fix incorrect usage of boost

### PR DESCRIPTION
**Pull request by mkirsano**

### CODE REVIEW

File: GeneratorInterface/Pythia6Interface/src/Pythia6Service.cc [hunk 1]
```diff
@@ -382,11 +382,18 @@ void Pythia6Service::setPYUPDAParams(bool afterPyinit)
 
 void Pythia6Service::setSLHAFromHeader( const std::vector<std::string> &lines )
 {
-
 	std::set<std::string> blocks;
 	unsigned int model = 0, subModel = 0;
+        bool useboost = true;
```
Comment (davidlt) [inline GeneratorInterface/Pythia6Interface/src/Pythia6Service.cc]: NAK. This is really not needed. This is a dead code which will be optimised out. Remove it. Stick with simple:
```
fnamest = std::string(boost::filesystem::unique_path().string());
fname = fnamest.c_str();
```
Then verify that it works (by running your workflows, valgrind, etc.)

Comment (davidlt) [inline GeneratorInterface/Pythia6Interface/src/Pythia6Service.cc]: Actually you don't `string()`, `.string()` already returns you `std::string`.

**Comment by mkirsano** [comment]
updated

**Comment by govoni** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**